sycl/source/detail/persistent_device_code_cache.cpp

```diff
@@ -664,6 +711,37 @@ PersistentDeviceCodeCache::getCompiledKernelFromDisc(
   return Binaries;
 }
 
+std::vector<char>
+PersistentDeviceCodeCache::getDeviceCodeIRFromDisc(const std::string &Key) {
+  std::vector<char> IR;
+
+  std::string DirName = getDeviceCodeIRPath(Key);
+
+  if (DirName.empty() || !OSUtil::isPathPresent(DirName))
+    return {};
+
+  std::string FileName = DirName + "/ir";
+  std::string FullFileName = FileName + ".bin";
+  if (OSUtil::isPathPresent(FullFileName)) {
+    if (!LockCacheItem::isLocked(FileName)) {
+      try {
+        IR = readBinaryDataFromFile(FullFileName);
+
+        // Explicitly update the access time of the file. This is required for
+        // eviction.
+        if (isEvictionEnabled())
+          saveCurrentTimeInAFile(FileName + CacheEntryAccessTimeSuffix);
+      } catch (...) {
+        // If read was unsuccessfull try the next item
```

Comment:
Is this comment correct? Looks more like we're just giving up and will compile fresh.

Reply:
Good catch, thanks!

Reply:
@cperkinsintel Just realised there's a potential bug in `getCompiledKernelFromDisc` (which I based `getDeviceCodeIRFromDisc` on): The iteration over the different `.bin` files doesn't make sense because for `kernel_compiler` as we don't write a source item to distinguish the binaries stored for the same cache key. `putCompiledKernelToDisc` does look for an unused filename to store the binary, but `getCompiledKernelFromDisc` will always attempt to read the `0.bin` file and use that unless an FS error occurs.